### PR TITLE
Allow case sensitivity in debugger command line

### PR DESCRIPTION
**Pull request by naerbnic**
# Description

The goal is to avoid uppercasing the Debugger command line until necessary, as future changes may depend on case sensitivity of arguments of commands.

Along the way, made some small local refactorings that also add case insensitivity to hex parsing.

Note that the command itself is still uppercased before it's checked against the possible commands, so the initial command itself is still case insensitive

# Manual testing

Manually tested the debug console in a -Denable_debugger=normal build. The observable result is that running an invalid debugger command line does not uppercase the whole line first. Other commands still run correctly, and variable references and hex values still operate.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [X] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [X] performed a self-review of my code.
- [X] commented on the particularly hard-to-understand areas of my code.
- [X] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [ ] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [X] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [X] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by johnnovak** [comment]
@naerbnic Thanks for the patience. Now is the time to finalise this, then the team will review it in the coming weeks.

**Review by japsmits — COMMENTED** [review]
This makes things easier to read. Thanks!

File: src/debug/debug.cpp [hunk 1]
```diff
@@ -988,43 +988,67 @@ static void SetCodeWinStart()
 /*    User input    */
 /********************/
 
-uint32_t GetHexValue(char* str, char*& hex)
+// Looks for the prefix string at the start of the initial string,
+// doing a case-insensitive compare. If it is found, consumes the
+// matching prefix and returns true. Otherwise, returns false and
+// leaves the string unchanged.
+static bool ConsumePrefixNoCase(char*& str, std::string_view prefix)
```
Comment (johnnovak) [inline src/debug/debug.cpp:995]: I know it's inconsistent in older code, but for new static functions we should use `lower_snake_case` as per our convention.

File: src/debug/debug.cpp [hunk 2]
```diff
@@ -988,43 +988,67 @@ static void SetCodeWinStart()
...
+{
+	for (size_t i = 0; i < prefix.size(); i++) {
+		if (!str[i]) return false;
```
Comment (johnnovak) [inline src/debug/debug.cpp:998]: Please run this through clang-format. `return false` should go into a separate line, enclosed with braces.

File: src/debug/debug.cpp [hunk 3]
```diff
@@ -988,43 +988,67 @@ static void SetCodeWinStart()
...
+		if (tolower(str[i]) != tolower(prefix[i]))
+			return false;
+	}
+
+	str = &str[prefix.size()];
+	return true;
+}
+
+static void StringToUpper(std::string& str)
```
Comment (johnnovak) [inline src/debug/debug.cpp:1007]: lower camel case

File: src/debug/debug.cpp [hunk 4]
```diff
@@ -988,43 +988,67 @@ static void SetCodeWinStart()
...
+// Looks for the prefix string at the start of the initial string,
+// doing a case-insensitive compare. If it is found, consumes the
+// matching prefix and returns true. Otherwise, returns false and
+// leaves the string unchanged.
+static bool ConsumePrefixNoCase(char*& str, std::string_view prefix)
+{
+	for (size_t i = 0; i < prefix.size(); i++) {
+		if (!str[i]) return false;
+		if (tolower(str[i]) != tolower(prefix[i]))
+			return false;
```
Comment (johnnovak) [inline src/debug/debug.cpp:1000]: enclose with braces

**Review by johnnovak — APPROVED** [review]
Looks fine @naerbnic , just some minor comments about naming.

**Comment by johnnovak** [comment]
Closing due to inactivity. Please feel free to raise another PR if you want to finish the work @naerbnic.

**Pull request closed without merging**